python/smashbox/utilities/__init__.py

```diff
@@ -41,500 +54,760 @@ def finalize_test():
     
     If exception is raised then smashbox terminates with non-zero exit code,
     """
+    global test_reporter
+    try:
+        test_reporter.finalize_test()
+    except:
+        pass
     d = make_workdir()
     scrape_log_file(d)
-
-######### HELPERS
-
-def reset_owncloud_account(reset_procedure=None, num_test_users=None):
-    """ 
-    Prepare the test account on the owncloud server (remote state). Run this once at the beginning of the test.
-
-    The reset_procedure defines what actually happens. If not set then the config default oc_account_reset_procedure
-    applies.
     
-    Normally the account is deleted and recreated ('delete')
+    sync_client_stop()  
 
-    If reset_procedure is set to 'keep' than the account is not deleted, so the state from the previous run is kept.
```

Comment:
I have also divided utilities into general functions, owncloud specific functions and general sync client functions like "make_workdir" etc.